src/surveys/components/ChoiceResponse.tsx

```diff
@@ -3,74 +3,67 @@ import { useMemo, useState } from 'preact/hooks';
 import { Button, Header } from '../../components';
 import {
   AnswerType,
-  ID,
   MultipleSettings,
   Question,
   QuestionOption,
   SingleSettings,
+  SurveyAnswer,
   Theme,
 } from '../../types';
+import { hexToRgba, shuffleArray } from '../../utils';
 
 interface SingleResponseProps {
   question: Question;
-  onAnswered: (questionId: ID, answerId: ID | ID[]) => void;
-  submitText?: string;
+  onAnswered: (answers: SurveyAnswer[]) => void;
   theme?: Theme;
+  submitText?: string;
 }
 
-export default function SelectResponse({
+export default function ChoiceResponse({
   question,
   onAnswered,
   submitText,
   theme,
 }: SingleResponseProps): VNode {
-  const [selectedOption, setSelectedOption] = useState<ID | ID[]>(
-    question.type === AnswerType.MULTIPLE ? [] : ''
-  );
+  const [selectedOption, setSelectedOption] = useState<QuestionOption[]>();
 
-  const handleOptionChange = (optionId: ID) => {
-    if ((selectedOption as ID[]).includes(optionId)) {
+  const handleOptionChange = (option: QuestionOption) => {
+    if (question.type === AnswerType.MULTIPLE) {
       setSelectedOption((prevState) =>
-        (prevState as ID[]).filter((id) => id !== optionId)
+        prevState?.includes(option)
+          ? prevState.filter((id) => id !== option)
+          : [...(prevState ?? []), option]
       );
     } else {
-      setSelectedOption((prevState) => [...(prevState as ID[]), optionId]);
+      setSelectedOption([option]);
     }
   };
 
   const handleSubmit = (e: h.JSX.TargetedEvent<HTMLFormElement, Event>) => {
     e.preventDefault();
-    if (!selectedOption) {
-      return;
-    }
-
-    onAnswered([{ answerId: selectedOption.id, answer: selectedOption.label }]);
+    if (!selectedOption) return;
+    onAnswered(
+      selectedOption.map((option) => ({
+        questionId: question.id,
```

Comment:
@afolasope Please remove the `questionId`, it's no longer needed.

Reply:
Okay